tests/test_compose.py

```diff
@@ -16,386 +15,463 @@
 OUTPUT_PATH = os.path.join('tests', 'output')
 
 
-class ComposeTest(unittest.TestCase):
```

Comment:
Switch to pure pytest style because pytest's parameterization feature does not work with `unittest.TestCase` subclasses: https://docs.pytest.org/en/latest/unittest.html#pytest-features-in-unittest-testcase-subclasses. Actually, we are mixing pytest and unittest in this project. I think we should migrate to only one of them, and I prefer pytest because it's more powerful.